consul/resource_consul_config_entry.go

```diff
@@ -16,6 +16,38 @@ func resourceConsulConfigEntry() *schema.Resource {
 		Update: resourceConsulConfigEntryUpdate,
 		Read:   resourceConsulConfigEntryRead,
 		Delete: resourceConsulConfigEntryDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), "/")
+				var kind, name, partition, namespace string
+				switch len(parts) {
+				case 2:
+					kind = parts[0]
+					name = parts[1]
+				case 4:
+					kind = parts[0]
+					name = parts[1]
+					partition = parts[2]
+					namespace = parts[3]
+				default:
+					return nil, fmt.Errorf(`expected path of the form "<kind>/<name>" or "<kind>/<name>/<partition>/<namespace>"`)
+				}
+
+				d.SetId(fmt.Sprintf("%s-%s", kind, name))
```

Comment:
Shouldn't partition and namespace be added to the ID to ensure uniqueness?

Reply:
The ID does not need to be unique. For config entries there is no ID that exist so we can use any placeholder there. We have other resources that not have a unique ID either, there is some examples at: https://github.com/hashicorp/terraform-provider-consul/pull/318#discussion_r1046452455